File: app/src/main/java/com/kidssaveocean/fatechanger/splashScreen/SplashScreenActivity.java
package com.kidssaveocean.fatechanger.splashScreen;

import android.os.Handler;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.WindowManager;

import com.kidssaveocean.fatechanger.R;

public class SplashScreenActivity extends AppCompatActivity {

    private final static int SPLASH_DURATION = 500;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        // defined theme in manifest: "@style/SplashTheme"
        // layout not needed

        new Handler().postDelayed(() -> {
            CheckOnboardingStatusTask checkOnboardingStatusTask = new CheckOnboardingStatusTask();
            checkOnboardingStatusTask.execute(this);
        },SPLASH_DURATION);
    }
}

File: app/src/main/java/com/kidssaveocean/fatechanger/splashScreen/SplashScreenActivity.kt
package com.kidssaveocean.fatechanger.splashScreen

import android.os.Bundle
import android.os.Handler
import android.os.Looper
import androidx.appcompat.app.AppCompatActivity
import kotlinx.coroutines.GlobalScope
import kotlinx.coroutines.launch

class SplashScreenActivity : AppCompatActivity() {

    //todo fix the damn splashscreen?
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        // defined theme in manifest: "@style/SplashTheme"
        // layout not needed

        Handler(Looper.getMainLooper()).postDelayed({
            GlobalScope.launch {
                CheckOnboardingStatusTask().execute(this@SplashScreenActivity)
            }
        }, SPLASH_DURATION.toLong())
    }

    companion object {

        private const val SPLASH_DURATION = 500
    }
}
